vars/withGCPEnv.groovy

```diff
@@ -81,6 +78,23 @@ def call(Map args = [:], Closure body) {
   }
 }
 
+/**
+* Read the vault secret and look for the required fields.
+* * credentials field is the one initially supported and kept for backward compatibility reasons.
+* * value field is the fallback field, this is the case used
+*/
+def readCredentialsContent(Map vaultSecretContent) {
+  def credentialsContent = vaultSecretContent?.credentials
+  if (credentialsContent?.trim()) {
+    return credentialsContent
```

Comment:
nit: out of curiosity I would drop a logger here to see which field is used :)

Reply:
See https://github.com/elastic/apm-pipeline-library/pull/1497/commits/121e3c3400d10f0c650f3de20cc7cbe4077cb798